### PR TITLE
use argparse for cli options

### PR DESCRIPTION
**Pull request by paul-krohn**
Precursor to passing values on the CLI for template tokens.

### CODE REVIEW

**Comment by plathrop** [comment]
Dear god. Yes, please. Probably worth an upstream pull request.

File: plumbum.py [hunk 1]
```diff
@@ -78,37 +79,26 @@ def lookup(instances, filter_by=None):
...
-        else:
-            # throw these away for now
-            extras.append(arg)
-
-    return template, namespace, region, filter_by, extras
+def interpret_options(args=sys.argv[1:]):
+
+    if '--version' in args:
+        print(__version__)
+        sys.exit()
```
Comment (plathrop) [inline plumbum.py:86]: Generally `sys.exit` from inside a function is no bueno, but given the rest of the code it seems fine here. Just keep in mind it's usually an anti-pattern.

File: plumbum.py [hunk 2]
```diff
@@ -78,37 +79,26 @@ def lookup(instances, filter_by=None):
...
-            key, value = arg.split('=', 2)
-            filter_by[key] = value
-        else:
-            # throw these away for now
-            extras.append(arg)
-
-    return template, namespace, region, filter_by, extras
+def interpret_options(args=sys.argv[1:]):
+
+    if '--version' in args:
```
Comment (plathrop) [inline plumbum.py:84]: Why not let `argparse` do this?

``` python
parser.add_argument("--version", help="Output version and exit", default=False, action="store_true")

if args.version:
    print(__version__)
    sys.exit()
```

File: plumbum.py [hunk 3]
```diff
@@ -78,37 +79,26 @@ def lookup(instances, filter_by=None):
...
-            extras.append(arg)
-        elif '=' in arg:
-            key, value = arg.split('=', 2)
-            filter_by[key] = value
-        else:
-            # throw these away for now
-            extras.append(arg)
-
-    return template, namespace, region, filter_by, extras
+def interpret_options(args=sys.argv[1:]):
```
Comment (plathrop) [inline plumbum.py:82]: Why the slice? Why not just `sys.argv`?

**Comment by paul-krohn** [comment]
Oh and thanks to @hanpeter for help with the tests.

File: plumbum.py (hunk 3 above)
Comment (paul-krohn) [inline plumbum.py:82]: It treats the program name as an argument if you do that.

**Pull request merged**